fix: Enabled detection of VS Code installed on macOS in Applications/ non PATH

devpool007 opened the pull request:
## TLDR

Updated editor.ts to enable option of detection of VS Code when installed in Applications and not installed code on the $PATH 

<!-- Add a brief description of what this pull request changes and why and any important things for reviewers to look at -->

## Dive Deeper

- As suggested by @markmcd '/Applications/Visual Studio Code.app/Contents/Resources/app/bin/code' was utilized as a fallback for checking if code command exists for VS Code Editor on macOS.

- Also editor.test.ts had to be updated since now what was expected as false before returned true especially in the 'should return false for vscode when not installed and not in sandbox mode', test for macOS/darwin 

<!-- more thoughts and in depth discussion here -->

## Reviewer Test Plan
1. Ether enable or disable sandbox by putting env variable GEMINI_SANDBOX=true or GEMINI_SANDBOX=false
2. npm run build
3. npm start
4. Type /editor and check the output 

In case of GEMINI_SANDBOX=true you should see this:
![Screenshot 2025-06-28 at 01 22 18](https://github.com/user-attachments/assets/dcd39f01-f600-4d08-b304-556f124dfa5c)

In case of GEMINI_SANDBOX=false you should see this:
![Screenshot 2025-06-28 at 00 56 05](https://github.com/user-attachments/assets/38f5cd07-c1a8-47eb-ac22-41750cf0c678)

<!-- when a person reviewes your code they should ideally be pulling and running that code. How would they validate your change works and if relevant what are some good classes of example prompts and ways they can exercise your changes -->

## Testing Matrix

<!-- Before submitting please validate your changes on as many of these options as possible -->

|          | 🍏  | 🪟  | 🐧  |
| -------- | --- | --- | --- |
| npm run  | ✅   | ✅   | ✅   |
| npx      | ✅   | ✅   | ✅   |
| Docker   | ✅   | ✅   | ✅   |
| Podman   | ❓  | -   | -   |
| Seatbelt | ✅   | -   | -   |

## Linked issues / bugs

<!-- Add links to any gh issues or other external bugs --->
Fixes #1695

NTaylorMullen commented on packages/core/src/utils/editor.ts:
```diff
@@ -41,6 +52,13 @@ export function checkHasEditorType(editor: EditorType): boolean {
     process.platform === 'win32' ? commandConfig.win32 : commandConfig.default;
+
+  // Special handling for VS Code on macOS
+  if (editor === 'vscode' && process.platform === 'darwin') {
```
Instead of this special casing would it be possible to build in the fallback logic into the editorCommands declaration?

devpool007 replied:
Hi, I think that's a nice idea would make more sense and code would be cleaner, I will implement this and update the code soon.

devpool007 commented:
Hi @NTaylorMullen I moved fallback logic inside editorCommands and updated code according to latest changes in main. I was also wondering should I also add macOSFallbacks for other editors too?
As right now in this PR the focus was on vscode but maybe it might be safe to do this for other editors too many of which are vscode based? 

Or should this be handled then in another PR? Lmk your thoughts :)

NTaylorMullen commented on packages/core/src/utils/editor.ts:
```diff
@@ -50,9 +59,15 @@ function commandExists(cmd: string): boolean {
 const editorCommands: Record<
   EditorType,
-  { win32: string[]; default: string[] }
+  { win32: string[]; default: string[]; macOSFallbacks?: string[] }
```
ditto elsewhere

```suggestion
  { win32: string[]; default: string[]; fallbackPaths?: string[] }
```

NTaylorMullen commented on packages/core/src/utils/editor.ts:
```diff
@@ -65,7 +80,24 @@ export function checkHasEditorType(editor: EditorType): boolean {
+  // On macOS, try fallback path for vscode available
+  if (
+    editor === 'vscode' &&
+    process.platform === 'darwin' &&
```
Lets not just check on darwin, lets check always, you can delete this bit and you should be able to delete the editor vscode check. SO fallback paths should be truly fall back paths and be checked always if they're configured

devpool007 replied:
Thanks for the suggestions, yes that makes sense actually, will do so.
